lib/recurly/Caster.js

```diff
@@ -17,33 +15,38 @@ function construct (Klass, data) {
  * @private
  * @param {Object} obj - The plain js object to cast to Recurly object
  * @param {Object} resourceDefs - The optional resource definitions used when casting. Only used for testing.
+ * @param {String} className - The optional className to use
  * @return {Object}
  */
-function castResponse (obj, resourceDefs = resources) {
-  const name = utils.className(obj)
+function castResponse (obj, resourceDefs = resources, className = null) {
+  const name = className || utils.className(obj.object)
   const Klass = resourceDefs[name]
 
   // either create a Klass object or just a plain js object
-  const resource = !(name && Klass) ? {} : construct(Klass, {})
+  // const resource = !(name && Klass) ? {} : construct(Klass, {})
```

Comment:
Do we think we're going to go back to this at some point?

Reply:
This is reminding me that i am actually missing some code here. Will pick it up tomorrow.